src/components/shared/CallNodeContextMenu.js

```diff
@@ -196,7 +197,27 @@ class CallNodeContextMenuImpl extends React.PureComponent<Props> {
     do {
       const funcIndex = callNodeTable.func[curCallNodeIndex];
       const stringIndex = funcTable.name[funcIndex];
-      stack += stringTable.getString(stringIndex) + '\n';
+      stack += stringTable.getString(stringIndex);
+      if (withFileLineColumnInfo) {
+        // Ported from src/profile-logic/profile-data.js' getOriginAnnotationForFunc()
+        let fileNameURL;
+        const fileNameIndex = funcTable.fileName[funcIndex];
+        if (fileNameIndex !== null) {
+          fileNameURL = stringTable.getString(fileNameIndex);
+          // Strip off any filename decorations from symbolication.
+          fileNameURL = parseFileNameFromSymbolication(fileNameURL).path;
+          const lineNumber = funcTable.lineNumber[funcIndex];
+          if (lineNumber !== null) {
+            fileNameURL += ':' + lineNumber;
+            const columnNumber = funcTable.columnNumber[funcIndex];
+            if (columnNumber !== null) {
+              fileNameURL += ':' + columnNumber;
+            }
+          }
+          stack += ` [${fileNameURL}]`;
+        }
+      }
+      stack += '\n';
```

Comment:
This definitely works fine. But what about using getOriginAnnotationForFunc directly? It looks like what we would like to do is pretty much exactly what this function is doing. We can remove this code duplicataion by using that function like so:
```js
const origin = getOriginAnnotationForFunc(
  funcIndex,
  funcTable,
  resourceTable,
  stringTable
);
```
then appending it to the function string. `resourceTable` is not available there yet, but you can get it from `thread`, [similar to getting `stringTable` and `funcTable` here](https://github.com/firefox-devtools/profiler/blob/be78e2170aa508be6964142a64053935c8b99e13/src/components/shared/CallNodeContextMenu.js#L189).

Reply:
I did this now but I wasn't 100% sure that the origin wouldn't also be returned so I added an extra optional parameter to `getOriginAnnotationForFunc` to make sure it's safe to call.